server/chat-plugins/mafia.ts

```diff
@@ -2626,32 +2704,40 @@ export const commands: Chat.ChatCommands = {
 				return this.errorReply(`You cannot add or remove players while IDEA roles are being picked.`); // needs to be here since eliminate doesn't pass the user
 			}
 			if (!target) return this.parse('/help mafia kill');
-			const player = game.playerTable[toID(target)];
-			const dead = game.dead[toID(target)];
-			let repeat;
-			if (dead) {
-				switch (cmd) {
-				case 'treestump':
-					repeat = dead.treestump && !dead.restless;
-					break;
-				case 'spirit':
-					repeat = !dead.treestump && dead.restless;
-					break;
-				case 'spiritstump':
-					repeat = dead.treestump && dead.restless;
-					break;
-				case 'kill': case 'kick':
-					repeat = !dead.treestump && !dead.restless;
-					break;
-				}
+			const player = game.getPlayer(toID(target));
+			if (!player) {
+				return this.errorReply(`${target.trim()} is not a player.`);
 			}
-			if (dead && repeat) return this.errorReply(`${dead.safeName} has already been ${cmd}ed.`);
-			if (player || dead) {
-				game.eliminate(toID(target), cmd);
-				game.logAction(user, `${cmd}ed ${(dead || player).safeName}`);
-			} else {
-				this.errorReply(`${target.trim()} is not a player.`);
+
+			let repeat, elimType;
+
+			switch (cmd) {
+			case 'treestump':
+				elimType = MafiaEliminateType.TREESTUMP;
+				repeat = player.isTreestump() && !player.isSpirit();
+				break;
+			case 'spirit':
+				elimType = MafiaEliminateType.SPIRIT;
+				repeat = !player.isTreestump() && player.isSpirit();
+				break;
+			case 'spiritstump':
+				elimType = MafiaEliminateType.SPIRITSTUMP;
+				repeat = player.isTreestump() && player.isSpirit();
+				break;
+			default:
+				if (cmd === 'kick') {
```

Comment:
Why is this not part of the main switch (cmd)?

Reply:
Probably because the original code had these in the same section. _shrug_. I'll change it.